libpod/container_top_linux.go

```diff
@@ -326,6 +331,7 @@ func (c *Container) execPS(psArgs []string) ([]string, bool, error) {
 
 	retryContainerExec := true
 	err = cmd.Run()
+	wPipe.Close()
```

Comment:
Just curious why you don't defer close this like you do rPipe?

Reply:
because that deadlocks
If you pass down a write fd you must close it in the parent if you want to wait for EOF on the reader in the parent. 
With defer the writer will be closed after we wait for EOF so that cannot work and deadlocks.